podcasts/UserSatisfactionSurveyManager.swift

```diff
@@ -0,0 +1,278 @@
+        hostingController.sizingOptions = .intrinsicContentSize
+        hostingController.presentationController?.delegate = self
+
+        if let sheet = hostingController.sheetPresentationController {
+            sheet.detents = [
+                .custom { context in
+                    let size = hostingController.sizeThatFits(in: CGSize(width: context.maximumDetentValue, height: .greatestFiniteMagnitude))
+                    return size.height
+                }
+            ]
+            sheet.prefersGrabberVisible = true
+            sheet.preferredCornerRadius = 24
+        }
+
+        source.present(hostingController, animated: true)
+        if !skipCheck {
+            Settings.addSurveyPresented()
+        }
+        Analytics.track(.userSatisfactionSurveyShown, properties: [
+            "trigger_event": event.rawValue,
+            "user_type": SubscriptionHelper.hasActiveSubscription() ? "plus" : "free"
+        ])
+    }
+
+    // MARK: - Helper Methods
+
+    private func hasUserLeftReview() -> Bool {
+        return !Settings.reviewRequestDates().isEmpty
+    }
+
+    private func hasShownSurveyRecently() -> Bool {
+        let surveyDates = Settings.surveyPresentationDates()
+        guard let lastSurveyDate = surveyDates.last else { return false }
+
+        let thirtyDaysAgo = Date().addingTimeInterval(-30 * 24 * 60 * 60)
+        return lastSurveyDate > thirtyDaysAgo
+    }
+
+    private func hasUserDeclinedRecently() -> Bool {
+        guard let lastDeclineDate = Settings.lastSurveyNotReallyDate() else { return false }
+
+        let sixtyDaysAgo = Date().addingTimeInterval(-60 * 24 * 60 * 60)
+        return lastDeclineDate > sixtyDaysAgo
+    }
+}
+
+extension UserSatisfactionSurveyManager: UIAdaptivePresentationControllerDelegate {
+    public func presentationControllerDidDismiss(_ presentationController: UIPresentationController) {
+        Analytics.track(.userSatisfactionSurveyDismissed, properties: [
+            "trigger_event": currentEvent?.rawValue ?? "unknown",
+            "user_type": SubscriptionHelper.hasActiveSubscription() ? "plus" : "free"
+        ])
+        currentEvent = nil
+    }
+}
+
+// MARK: - AnalyticsAdapter
+
+extension UserSatisfactionSurveyManager: AnalyticsAdapter {
+    public func track(name: String, properties: [AnyHashable: Any]?) {
+        guard let analyticsEvent = mapAnalyticsEventToSurveyTrigger(name: name) else { return }
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1) { [weak self] in
+            guard let self = self else { return }
+
+            if self.shouldShowSurvey(for: analyticsEvent) {
+                guard let topViewController = SceneHelper.rootViewController() else { return }
+                self.presentSurvey(from: topViewController, event: analyticsEvent)
+            }
+        }
+    }
+
+    private func mapAnalyticsEventToSurveyTrigger(name: String) -> SurveyTriggerEvent? {
+        switch name {
+        case "episode_starred":
```

Comment:
Instead of using the event name directly here, what do you think of using the enum rawValue like this:

`.episodeStarred.rawValue.toSnakeCaseFromCamelCase()` 

this way if in the future someone changes the events name or removes it they will warned of this dependency.